projects/valtimo/dashboard-management/src/lib/components/dashboard-details/dashboard-details.component.ts

```diff
@@ -1,11 +1,114 @@
-import {Component, OnInit} from '@angular/core';
+import {AfterViewInit, Component, TemplateRef, ViewChild, ViewEncapsulation} from '@angular/core';
+import {ActivatedRoute} from '@angular/router';
+import {BehaviorSubject, combineLatest, map, Observable, tap} from 'rxjs';
+import {DashboardItem, DashboardWidget, WidgetModalType} from '../../models';
+import {dashboardListMock, widgetListMock} from '../../mocks';
+import {
+  CarbonTableConfig,
+  ColumnType,
+  createCarbonTableConfig,
+  PageTitleService,
+} from '@valtimo/components';
+import {TranslateService} from '@ngx-translate/core';
 
 @Component({
   templateUrl: './dashboard-details.component.html',
   styleUrls: ['./dashboard-details.component.scss'],
+  encapsulation: ViewEncapsulation.None,
 })
-export class DashboardDetailsComponent implements OnInit {
-  constructor() {}
+export class DashboardDetailsComponent implements AfterViewInit {
+  @ViewChild('moveButtonsTemplate', {static: false}) moveButtonsTemplate: TemplateRef<any>;
 
-  ngOnInit(): void {}
+  public modalType: WidgetModalType = 'create';
+  public tableConfig!: CarbonTableConfig;
+  public readonly currentDashboard$: Observable<DashboardItem> = combineLatest([
+    this.route.params,
+    this.translateService.stream('key'),
+  ]).pipe(
+    map(([params]) => dashboardListMock.find(mockItem => mockItem.key === params.id)),
+    tap(currentDashboard => {
```

Comment:
If you know the type of currentDashboard, please add it

Reply:
I think the observable on line 24 has a type defined?